rasterio/plot.py

```diff
@@ -50,6 +50,9 @@ def show(source, cmap='gray', with_bounds=True):
         rather than pixel coordinates. Only works when source is
         (raster dataset, bidx).
     """
+    if plt is None:  # pragma: no cover
+        raise ImportError("Could not import matplotlib")
+
```

Comment:
In other words, while we prefer to keep our imports at the top of the file `matplotlib` is one situation where it's okay to import inside a function.